### ISSUE TITLE
Unable to update Security Group If it is associated with an Network Interface.

### ISSUE THREAD

**Issue by DanyangLinda** [Azure/azure-sdk-for-java#1259]
I created an Network with a subnet associating an Security Group. The Network and subnet was used to create an Network Interface. Then I call 
```
final NetworkSecurityGroup networkSecurityGroup = azure.networkSecurityGroups().getByGroup(resourceGroupName, networkSecurityGroupName);

networkSecurityGroup.update().withoutRule(rule).apply();
```

I get the following Error:

```
java.lang.IllegalArgumentException: networkInterfaces.location is required and cannot be null.
	at com.microsoft.rest.Validator.validate(Validator.java:105)
	at com.microsoft.azure.management.network.implementation.NetworkSecurityGroupsInner.createOrUpdateWithServiceResponseAsync(NetworkSecurityGroupsInner.java:468)
networkInterfaces.location is required and cannot be null.
	at com.microsoft.azure.management.network.implementation.NetworkSecurityGroupsInner.createOrUpdateAsync(NetworkSecurityGroupsInner.java:436)
	at com.microsoft.azure.management.network.implementation.NetworkSecurityGroupImpl.createInner(NetworkSecurityGroupImpl.java:165)
	at com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.GroupableParentResourceImpl.createResourceAsync(GroupableParentResourceImpl.java:53)
	at com.microsoft.azure.management.resources.fluentcore.model.implementation.CreatableUpdatableImpl.updateResourceAsync(CreatableUpdatableImpl.java:189)
	at com.microsoft.azure.management.resources.fluentcore.model.implementation.CreateUpdateTask.executeAsync(CreateUpdateTask.java:43)
	at com.microsoft.azure.TaskGroupBase.executeAsync(TaskGroupBase.java:79)
	at com.microsoft.azure.management.resources.fluentcore.model.implementation.CreatableUpdatableImpl.executeTaskGroupAsync(CreatableUpdatableImpl.java:196)
	at com.microsoft.azure.management.resources.fluentcore.model.implementation.CreatableUpdatableImpl.applyAsync(CreatableUpdatableImpl.java:157)
	at com.microsoft.azure.management.resources.fluentcore.model.implementation.CreatableUpdatableImpl.apply(CreatableUpdatableImpl.java:178)
	at com.microsoft.azure.management.compute.samples.ManageVirtualMachine.main(ManageVirtualMachine.java:312)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

**Comment by DanyangLinda** [comment]
I found that if the Network Resource Group associates with an Network Interface, `networkSecurityGroup.inner().networkInterfaces()` is not `null` but a list containing a NetworkInterfaceInner, and all the filed of the NetworkInterfaceInner object are `null`. So to make it work, I have to manually set up the location filed of the NetworkInterfaceInner object: 

```
networkSecurityGroup2.inner().networkInterfaces().get(0).withLocation("westus")
```

I think this is really a bug.

**Comment by nvonstein** [comment]
This bug was also addressed here #935 and fixed in 1.0.0-beta2. However, I encounter the same issue again in beta3 & beta4.

**Comment by martinsawicki** [comment]
fixed for beta5.
(this [test case](https://github.com/Azure/azure-sdk-for-java/blob/master/azure/src/test/java/com/microsoft/azure/management/TestNSG.java) has been modified to involve a NIC and is passing)
